### PR TITLE
not using spres_lower_limit limit non-top segments

### PR DESCRIPTION
**Pull request by GitPaean**
with multi-segments, the segment pressure can be lower than the bhp, so it is not desirable to use spres_lower_limit to limit the segment pressure. We only use spres_lower_limit to limit the bhp, which is still valid until we begin using bhp limit lower than 1 bar.

### CODE REVIEW

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/simulators/wells/MultisegmentWellPrimaryVariables.cpp [hunk 1]
```diff
@@ -183,10 +183,13 @@ updateNewton(const BVectorWell& dwells,
...
-            // TODO: we might need to distinguish SPres of the top segment from the SPres of the other segments, since only the top segment
-            // is involved in the well constraints/control checking
-            constexpr double spres_lower_limit = 1. * unit::barsa - 1. * unit::Pascal;
-            value_[seg][SPres] = std::max(old_primary_variables[seg][SPres] - dx_limited, spres_lower_limit);
+            if (seg == 0) { // top segment pressure is the bhp
+                constexpr double bhp_lower_limit = 1. * unit::barsa - 1. * unit::Pascal;
+                value_[seg][SPres] = std::max(old_primary_variables[seg][SPres] - dx_limited, bhp_lower_limit);
+            } else {
+                // enforcing 0. as the lower limit for the segment pressure of other segment
+                value_[seg][SPres] = std::max(old_primary_variables[seg][SPres] - dx_limited, 0.);
```
Comment (GitPaean) [inline opm/simulators/wells/MultisegmentWellPrimaryVariables.cpp:191]: we should discuss whether even keeping 0. as the lower limit. or removing this limit and give a warning message when segment pressure goes below 0.0. @steink

**Comment by GitPaean** [comment]
jenkins build this please

**Comment by GitPaean** [comment]
jenkins build this please

File: opm/simulators/wells/MultisegmentWellPrimaryVariables.cpp [hunk 2]
```diff
@@ -183,10 +186,21 @@ updateNewton(const BVectorWell& dwells,
...
-            value_[seg][SPres] = std::max(old_primary_variables[seg][SPres] - dx_limited, spres_lower_limit);
+            if (seg == 0) { // top segment pressure is the bhp
+                constexpr double bhp_lower_limit = 1. * unit::barsa - 1. * unit::Pascal;
+                value_[seg][SPres] = std::max(old_primary_variables[seg][SPres] - dx_limited, bhp_lower_limit);
+            } else {
+                // enforcing 0. as the lower limit for the segment pressure of other segment
+                const double newton_update_value = old_primary_variables[seg][SPres] - dx_limited;
+                if (newton_update_value >= 0.) {
+                    value_[seg][SPres] = newton_update_value;
+                } else {
```
[the remark this answers is not in the document]
Comment (GitPaean) [reply opm/simulators/wells/MultisegmentWellPrimaryVariables.cpp:197]: It is a good idea. Then the debug info we need to output somewhere else. Will do that before the PR goes in.

**Comment by GitPaean** [comment]
The PR got confirmed to fix the targeting field case running. Will finalize the PR and try to get it merged.

File: opm/simulators/wells/MultisegmentWellPrimaryVariables.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (GitPaean) [reply opm/simulators/wells/MultisegmentWellPrimaryVariables.cpp]: proposed some way to do this, please have a look.

**Comment by GitPaean** [comment]
I am marking this PR as ready for review.

**Comment by GitPaean** [comment]
jenkins build this please

**Review by steink — APPROVED** [review]
Looks good to me

**Comment by GitPaean** [comment]
With the approval, and jenkins regression test passing, I am getting this PR in now.

**Pull request merged**